### PR TITLE
Make SettlementEncoder aware of internalizable interactions

### PR DESCRIPTION
**Pull request by fleupold**
Tiny part of #635 

This PR leaks the concept of which orders are internalizable from the HTTP solver into the settlement encoder. This is needed so that we can log, simulate and record the original solutions (including interactions that get internalized) for solvers to inspect (solvers can lie about amounts and play other games on those interactions so this is needed for transparency).

I didn't realize that we are not correctly sorting these interactions in the HTTP solver (in fact they don't have coordinates on them), but in most cases there is only a single interaction so just logging it in whatever order we currently have it should sufficient (also the changes in this PR wouldn't be affected by fixing the ordering).

A next step would be to log these unoptimized settlements and potentially also update them to the solution competition endpoint.

### Test Plan

CI (no user visible change yet)

### CODE REVIEW

File: crates/solver/src/settlement/settlement_encoder.rs
Comment (josojo) [inline crates/solver/src/settlement/settlement_encoder.rs:21]: Have you considered defining another function on the trait Interaction that leaks whether it's private?
```
pub trait Intearction {
  fn encode() ->{};
  fn is_private_interaction(&self) -> bool
}
```
The advantage would be that we don't have to hand over tuples all the time. But I am not sure whether this abstraction is really well suited for all Interaction types and it could be too general. I have not thought deeply about it.

**Review by vkgnosis — APPROVED** [review]

**Comment by fleupold** [comment]
@cowprotocol/backend could I please get a second review?

File: crates/solver/src/settlement/settlement_encoder.rs
Comment (josojo) [reply crates/solver/src/settlement/settlement_encoder.rs:21]: Hey Felix, 
I did not approve yet, as I wanted someone else opinion
.
>My worry is that this logic would have to be implemented on every interaction (and someone adding a new interaction might forget to add support for internalizing)

If it's part of the trait the compile will tell them. I don't see this as a compelling reason.
But I think it's the right approach, given that we everywhere in the code keep the order information in the execution plans and they are separate from the interaction themselves.

Comment (josojo) [reply crates/solver/src/settlement/settlement_encoder.rs:21]: I think there is also a spelling error in the Name: MaybeInternalizabelInteraction 
Should it be MaybeInternalizableInteraction?

Maybe it's also wise to leave out the Maybe from the name, as the bool is strictly determining whether it's internalizable?

Comment (vkgnosis) [reply crates/solver/src/settlement/settlement_encoder.rs:21]: Felix' reasoning makes sense to me. I think it is fine the way it is in the Pr now.

**Review by josojo — APPROVED** [review]

**Pull request merged**
